source/server/config_validation/server.h

```diff
@@ -107,6 +107,9 @@ class ValidationInstance final : Logger::Loggable<Logger::Id::main>,
     return validation_context_;
   }
   Configuration::ServerFactoryContext& serverFactoryContext() override { return server_context_; }
+  Configuration::TransportSocketFactoryContext& transportSocketFactoryContext() override {
+    return server_context_;
```

Comment:
Maybe `server_context_` is not a `transportSocketFactoryContext`. Instead, create another server life long object inside/beside `server_context_`

Reply:
With this change `server_context_` (`ServerFactoryContextImpl`) is both a `ServerFactoryContext` and `TransportSocketFactoryContext`; maybe renaming `server_context_` and `ServerFactoryContextImpl` to something else might help the confusion? Sharing the implementation makes this change overhead-free, but I'm not opposed to separating them out either, if it is too confusing.